src/core/Xlang.ml

```diff
@@ -41,6 +41,19 @@ let lang_of_opt_xlang_exn (x : t option) : Lang.t =
   | None -> failwith (Lang.unsupported_language_message "unset")
   | Some xlang -> to_lang_exn xlang
 
+let is_compatible ~require ~provide =
+  match (require, provide) with
+  | LRegex, LRegex
```

Comment:
I wonder how do we assign `LRegex` or `LSpacegrep` or `LAliengrep` to a target file?

Reply:
There are target selectors (e.g. "select all C files" or "select all text files") and there are analyzers (e.g. "match a C pattern" or "match a regexp"). These are two concepts that are mixed up in the rule's `languages` field and other places. I'm continuing efforts to make the distinction clear where possible but the `languages` field is probably here to stay for compatibility reasons. It is parsed into the xlang type which couples a selector and an analyzer.

Your question is:
> I wonder how do we assign `LRegex` or `LSpacegrep` or `LAliengrep` to a target file?

These 3 analyzers should be compatible more or less with any text file. For Spacegrep, we may be more specific and restrict ourselves to text that looks like a human-readable program. To make this clear and explicit, we should provide target selectors named Text and Program. Right now, we have no such thing. Spacegrep does its own filtering i.e. if you scan a binary file with Spacegrep, it will count as scanned but Spacegrep will detect that it's a binary file and won't parse it and will return zero matches without an error. To make things clear, a rule should have two fields:
```
# At least one target selector must be specified.
# Having just one target selector is the most common case.
# The union of selected targets will be used.
target-selectors:
- javascript
- typescript
analyzer: typescript
```
Here's another example where we would scan C files (`.c`, `.h`) with a regexp:
```
target-selectors:
- c
analyzer: regex
```
It's currently not possible without specifying `include: *.c` and `include: *.h` in the `paths` section of the rule which is error-prone and limited to files that can be selected by name, e.g. there's no way to select Bash scripts.

I encourage everyone to start thinking in terms of selectors and analyzers.